libftl/ingest.c

```diff
@@ -92,6 +92,96 @@ OS_THREAD_ROUTINE _ingest_get_rtt(void *data) {
     return 0;
 }
 
+ftl_status_t find_closest_available_ingest(const char* ingestHosts[], int ingestsCount, char* bestIngestHostComputed)
+{
+    ftl_ingest_t* ingestElements;
+    int i;
+
+    if ((ingestElements = malloc(sizeof(ftl_ingest_t) * ingestsCount)) == NULL) {
+        return FTL_MALLOC_FAILURE;
+    }
+
+    for (i = 0; i < ingestsCount; i++) {
+        size_t host_len = strlen(ingestHosts[i]);

```

Comment:
I believe strlen doesn't include the /0. I think you need to add a +1 and make sure your string is terminated with a /0

Reply:
Good catch!

Reply:
Can you run the sdk / its tests with valgrind? It would have caught this easily 🙂

Reply:
@connor4312 I will try to do that. It seems valgrind doesn't support well on Mac. Is there any test cases or I just run the sample app.

Reply:
Ah, yea, unfortunately valgrind is pretty unhappy on OSX. I'd use a VM or VirtualBox and run it there. You could also have this repository configured to run CI in Travis, and run tests with valgrind in Travis.